runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterOverlay.java

```diff
@@ -97,8 +97,14 @@ public Dimension render(Graphics2D g)
 						(int) DIAMETER, (int) DIAMETER);
 				}
 			}
-
-			renderRegen(g, ComponentID.MINIMAP_SPEC_ORB, plugin.getSpecialPercentage(), SPECIAL_COLOR);
+			if (!plugin.wearingSoulreaperAxe)
```

Comment:
This would cause a player wearing the axe with `config.showSoulStack()` to have nothing rendered. I think this check should instead be:

```java
if (!plugin.wearingSoulreaperAxe || !config.showSoulStack())
{
	// ...
}
else
{
	// ...
}
```

Reply:
i'm not sure if i follow correctly, but the situation you're describing shows the stack decay without the changes.